pennylane/circuit_graph.py

```diff
@@ -12,307 +12,287 @@
 
         Returns:
-            list (int): all operation indices on the wire,
-            in temporal order
+            list[int]: indices of operations on the wire, in temporal order
         """
-        return [op.idx for op in self._grid[wire]]
+        return [op.queue_idx for op in self._grid[wire]]
 
     def ancestors(self, ops):
-        """Returns all ancestor operations of a given set of operations.
+        """Ancestor operations of a given set of operations.
 
         Args:
-            ops (Iterable[int]): a given set of operations labelled by integer
-                position in the queue
+            ops (Iterable[Operation]): set of operations in the circuit
 
         Returns:
-            set[int]: integer position of all operations
-            in the queue that are ancestors of the given operations
+            set[Operation]: ancestors of the given operations
         """
-        subGs = [self.graph.subgraph(nx.dag.ancestors(self.graph, o)) for o in ops]
-        return set().union(*[set(subG.nodes()) for subG in subGs]) - set(ops)
+        return set().union(*(nx.dag.ancestors(self._graph, o) for o in ops)) - set(ops)
 
     def descendants(self, ops):
-        """Returns all descendant operations of a given set of operations.
+        """Descendant operations of a given set of operations.
 
         Args:
-            ops (Iterable[int]): a given set of operations labelled by integer
-                position in the queue
+            ops (Iterable[Operation]): set of operations in the circuit
 
         Returns:
-            set[int]: integer position of all operations
-            in the queue that are descendants of the given operations
+            set[Operation]: descendants of the given operations
         """
-        subGs = [self.graph.subgraph(nx.dag.descendants(self.graph, o)) for o in ops]
-        return set().union(*[set(subG.nodes()) for subG in subGs]) - set(ops)
+        return set().union(*(nx.dag.descendants(self._graph, o) for o in ops)) - set(ops)
 
-    def get_nodes(self, ops):
-        """Given a set of operation indices, return the nodes corresponding to the indices.
+    def _in_topological_order(self, ops):
+        """Sorts a set of operations in the circuit in a topological order.
 
         Args:
-            ops (Iterable[int]): a given set of operations labelled by integer
-                position in the queue
+            ops (Iterable[Operation]): set of operations in the circuit
 
         Returns:
-            List[Node]: nodes corresponding to given integer positions in the queue
+            Iterable[Operation]: same set of operations, topologically ordered
         """
-        return [self.graph.nodes[i] for i in ops]
+        G = nx.DiGraph(self._graph.subgraph(ops))
+        return nx.dag.topological_sort(G)
 
-    def get_ops(self, ops):
-        """Given a set of operation indices, return the operation objects.
+    def ancestors_in_order(self, ops):
```

Comment:
This method (and descendents in order) could simply be moved as keyword arguments to their parent methods:
```python
def ancestors(self, ops, topological_order=False):
```

Reply:
That would change the return type, which is why it might be good to keep them separate. `ancestors` returns a set, `ancestors_in_order` returns a list (could also return an iterable).

Reply:
I would suggest that both this method, `descendants_in_order`, and `_by_idx` be combined into a single method called `sort_topologically` which takes a set and returns a sorted set, no matter if it's ancestors or descendants.

Reply:
I'll try it out and see how it would work while I'm working on the QNode refactor. The main issue IMO is lack of brevity,
`self.circuit.sort_topologically(self.circuit.ancestors(ops))` vs. `self.circuit.ancestors_in_order(ops)`